pkg/kubelet/kubelet.go

```diff
@@ -2176,6 +2187,9 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 			// TODO: Do we want to support this?
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
+
+		kl.sourcesReady.AddSource(u.Source)
```

Comment:
The reason for moving Addsource() after updating the podManager is to make sure when checking sourcesReady, the podManager is already updated. The change is needed because if update fails, we should not say sources are already ready. Volume reconstruct process and orphaned pods cleanup depends on this checking to make sure all sources are ready before the clean up.